Tests/AblyChatTests/DefaultRoomPresenceTests.swift

```diff
@@ -0,0 +1,452 @@
+import Ably
+@testable import AblyChat
+import Testing
+
+struct DefaultRoomPresenceTests {
+    // MARK: CHA-PR3
+
+    // @spec CHA-PR3a
+    // @spec CHA-PR3e
+    @Test
+    func usersMayEnterPresence() async throws {
+        // Given
+        let realtimePresence = MockRealtimePresence()
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let featureChannel = MockFeatureChannel(channel: channel, resultOfWaitToBeAblePerformPresenceOperations: .success(()))
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "client1", logger: TestLogger())
+
+        // When
+        try await defaultPresence.enter(data: ["status": "Online"])
+
+        // Then
+        #expect(realtimePresence.callRecorder.hasRecord(
+            matching: "\(#selector(MockRealtimePresence.enterClient(_:data:callback:)))",
+            arguments: ["name": "client1", "data": ["status": "Online"].toAblyCocoaData()])
```

Comment:
This assertion is very confusing. It _reads_ like you're asserting that the data passed to ably-cocoa is `["status": "Online"]`, which doesn't match the data structure that CHA-PR2 tells us should be passed to ably-cocoa.

I'd suggest being explicit here: one of either

```suggestion
arguments: ["name": "client1", "data": ["userCustomData": ["status": "Online"]]]
```

or

```suggestion
arguments: ["name": "client1", "data": PresenceDataDTO(userCustomData: ["status": "Online"]).toJSONObject.toAblyCocoaDataDictionary]
```

(I think that the first one is probably more readable even though you could argue that there's some overlap with the unit tests for `PresenceDataDTO`)

Reply:
done